### PR TITLE
Add ArrayVarReductionAux

### LINKED ISSUE #22809: Add an AuxKernel that computes the sum of the components of an ArrayVar
## Reason

To be able to compute the sum of the components of an ArrayVar. This is a useful capability for a
large number of applications.

## Design

Add a new object: `ArrayVarSumAux`

## Impact

none

### PR DESCRIPTION
**Pull request by snschune**
closes #22809 

## Reason

Compute min/max/sum/average of array variables and store it in standard field variable.

## Design

AuxKernel

## Impact

None.

### CODE REVIEW

**Comment by snschune** [comment]
@YaqiWang fyi

**Comment by YaqiWang** [comment]
The failure does not seem related. @lindsayad can you find someone review this PR?

File: framework/src/auxkernels/ArrayVarReductionAux.C [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        r += _array_variable[_qp](t);
+      break;
+
+    case 3:
+      r = 0;
+      for (unsigned int t = 0; t < _nentries; ++t)
+        r += _array_variable[_qp](t) / _nentries;
+      return r;
+  }
+  return r;
```
Comment (zachmprince) [inline framework/src/auxkernels/ArrayVarReductionAux.C:68]:
```suggestion
  switch (_value_type)
  {
    case 0:
      return _array_variable[_qp].maxCoeff();

    case 1:
      return _array_variable[_qp].minCoeff();

    case 2:
      return _array_variable[_qp].sum();

    case 3:
      return _array_variable[_qp].mean();
  }
  
  return 0.0;
```

File: framework/src/auxkernels/ArrayVarReductionAux.C [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      MooseEnum("max=0 min=1 sum=2 average=3", "sum"),
+      "Type of reduction operation. Options are max, min, sum, and average.");
+  params.addClassDescription("Takes an array variable and performs a reduction operation on it "
+                             "(max, min, sum, average) and stores as a standard variable.");
+  return params;
+}
+
+ArrayVarReductionAux::ArrayVarReductionAux(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _nentries(getArrayVar("array_variable", 0)->count()),
```
Comment (zachmprince) [inline framework/src/auxkernels/ArrayVarReductionAux.C:30]: This member isn't necessary with my suggestion.

**Review by zachmprince — CHANGES_REQUESTED** [review]
Suggestion for less code.

File: framework/include/auxkernels/ArrayVarReductionAux.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#include "AuxKernel.h"
+
+class ArrayVarReductionAux : public AuxKernel
+{
+public:
+  static InputParameters validParams();
+  ArrayVarReductionAux(const InputParameters & parameters);
+
+protected:
+  virtual Real computeValue();
```
Comment (lindsayad) [inline framework/include/auxkernels/ArrayVarReductionAux.h:21]:
```suggestion
  Real computeValue() override;
```

File: framework/include/auxkernels/ArrayVarReductionAux.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  /// variable for storing the array variable that needs to be processed
```
Comment (lindsayad) [inline framework/include/auxkernels/ArrayVarReductionAux.h:23]:
```suggestion
  /// the array variable value we will perform the reduction on
```

File: test/tests/auxkernels/array_var_reduction/array_var_reduction.i [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+  [max_var]
+  []
+
+  [sum_var]
+  []
+
+  [average_var]
+  []
+
```
Comment (lindsayad) [inline test/tests/auxkernels/array_var_reduction/array_var_reduction.i:31]:
```suggestion
  [min_var][]
  [max_var][]
  [sum_var][]
  [average_var][]
```

**Review by lindsayad — CHANGES_REQUESTED** [review]
looks good!

**Review by lindsayad — APPROVED** [review]

**Review by zachmprince — APPROVED** [review]

**Pull request merged**